src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,120 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * Loads tasks and saves tasks in the specified file.
+ *
+ * @author Lim Ai Lin
+ */
+public class Storage {
+
+    private enum Commands {
+        DEADLINE,
+        TODO,
+        EVENT
+    }
+
+    private final File FILE;
+    private final String FILE_PATH;
+
+    /**
+     * Creates a storage object to load and save all tasks.
+     *
+     * @param filePath The text file to load the tasks from and save the tasks to.
+     */
+    public Storage(String filePath) {
+        this.FILE_PATH = filePath;
+        this.FILE = new File(filePath);
+    }
+
+    protected ArrayList<Task> load() throws IOException {
+        ArrayList<Task> ls = new ArrayList<>(100);
+        try {
+            if (this.FILE.createNewFile()) {
+                System.out.println("Dino created a new file: " + FILE.getName() + "\n");
+            } else {
+                BufferedReader br = new BufferedReader(new FileReader(FILE));
+                try {
+                    String current = br.readLine();
+                    while (current != null) {
+                        String[] str = current.split("\\|", 3);
+                        Commands myTask = Commands.valueOf(str[0].toUpperCase(Locale.ROOT));
+                        switch (myTask) {
+                        case DEADLINE:
+                            String[] dl = str[2].split("\\|", 2);
+                            Task deadline = new Deadline(dl[0], dl[1]);
+                            ls.add(deadline);
+                            if (Objects.equals(str[1], "1")) {
+                                deadline.markAsDone();
+                            }
+                            break;
+                        case TODO:
+                            Task todo = new ToDo(str[2]);
+                            ls.add(todo);
+                            if (Objects.equals(str[1], "1")) {
+                                todo.markAsDone();
+                            }
+                            break;
+                        case EVENT:
+                            String[] evnt = str[2].split("\\|", 2);
+                            Task event = new Event(evnt[0], evnt[1]);
+                            ls.add(event);
+                            if (Objects.equals(str[1], "1")) {
+                                event.markAsDone();
+                            }
+                            break;
+                        default:
+                            Ui.invalidTask();
+                        }
+                        current = br.readLine();
+                    }
```

Comment:
I think here is a little hard to understand. Maybe can consider the use of guard clause.